diff --git a/node_modules/@react-native-masked-view/masked-view/types/index.d.ts b/node_modules/@react-native-masked-view/masked-view/types/index.d.ts
index acafcc2..44bcadc 100644
--- a/node_modules/@react-native-masked-view/masked-view/types/index.d.ts
+++ b/node_modules/@react-native-masked-view/masked-view/types/index.d.ts
@@ -11,8 +11,5 @@ interface MaskedViewProps extends ReactNative.ViewProps {
  * @see https://github.com/react-native-masked-view/masked-view
  */
 declare class MaskedViewComponent extends React.Component<MaskedViewProps> {}
-declare const MaskedViewBase: ReactNative.Constructor<
-  ReactNative.NativeMethods
-> &
-  typeof MaskedViewComponent;
+declare const MaskedViewBase: typeof MaskedViewComponent;
 export default class MaskedView extends MaskedViewBase {}
