diff --git a/node_modules/react-native-keys/android/build.gradle b/node_modules/react-native-keys/android/build.gradle
index 7d07aae..e01d9f4 100644
--- a/node_modules/react-native-keys/android/build.gradle
+++ b/node_modules/react-native-keys/android/build.gradle
@@ -155,6 +155,7 @@ android {
     excludes = [
       "**/libc++_shared.so",
       "**/libfbjni.so",
+      "**/libcrypto.so",
       "**/libreactnativejni.so",
       "**/libjsi.so",
       "**/libreact_nativemodule_core.so",
diff --git a/node_modules/react-native-keys/src/util/common.js b/node_modules/react-native-keys/src/util/common.js
index 4f26e10..b671a59 100644
--- a/node_modules/react-native-keys/src/util/common.js
+++ b/node_modules/react-native-keys/src/util/common.js
@@ -75,7 +75,7 @@ module.exports.genTSType = (allKeys) => {
   Object.keys(allKeys?.secure ?? {}).forEach((key) => {
     result += `\n  ${key}: string;`;
   });
-  result += '\n[key: string]: string;\n};\n';
+  result += '\n};\n';
   fs.outputFileSync(path.join(SRC_PATH, 'type.ts'), result);
 };
 
