diff --git a/node_modules/react-native-animateable-text/android/src/main/java/com/reactnativereanimatedtext/JBTextViewManager.java b/node_modules/react-native-animateable-text/android/src/main/java/com/reactnativereanimatedtext/JBTextViewManager.java
index ccc81bd..83dd64a 100644
--- a/node_modules/react-native-animateable-text/android/src/main/java/com/reactnativereanimatedtext/JBTextViewManager.java
+++ b/node_modules/react-native-animateable-text/android/src/main/java/com/reactnativereanimatedtext/JBTextViewManager.java
@@ -8,7 +8,9 @@
 package com.reactnativereanimatedtext;
 
 import android.content.Context;
+import android.os.Build;
 import android.text.Spannable;
+import android.view.Gravity;
 import androidx.annotation.Nullable;
 import com.facebook.react.bridge.ReadableMap;
 import com.facebook.react.bridge.ReadableNativeMap;
@@ -105,13 +107,18 @@ public class JBTextViewManager
     int textBreakStrategy =
       TextAttributeProps.getTextBreakStrategy(paragraphAttributes.getString("textBreakStrategy"));
 
+    int currentJustificationMode = 0;
+    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+      currentJustificationMode = view.getJustificationMode();
+    }
+
     return new ReactTextUpdate(
       spanned,
       state.hasKey("mostRecentEventCount") ? state.getInt("mostRecentEventCount") : -1,
       false, // TODO add this into local Data
-      TextAttributeProps.getTextAlignment(props, TextLayoutManager.isRTL(attributedString)),
+      TextAttributeProps.getTextAlignment(props, TextLayoutManager.isRTL(attributedString), Gravity.LEFT),
       textBreakStrategy,
-      TextAttributeProps.getJustificationMode(props));
+      TextAttributeProps.getJustificationMode(props, currentJustificationMode));
   }
 
   @Override
